src/discussions/data/hooks.js

```diff
@@ -174,3 +178,16 @@ export const useCurrentDiscussionTopic = () => {
   }
   return null;
 };
+
+export const useUserPrivilage = () => {
+  const isUserAdmin = useSelector(selectUserIsStaff);
+  const userHasModerationPrivilages = useSelector(selectUserHasModerationPrivileges);
+  const isUserGroupTA = useSelector(selectUserIsGroupTa);
+  const isCourseAdmin = useSelector(selectIsCourseAdmin);
+  const isCourseStaff = useSelector(selectIsCourseStaff);
+
+  return (
+    handleAddPostForRoles(isUserAdmin, userHasModerationPrivilages,
+      isUserGroupTA, isCourseAdmin, isCourseStaff)
+  );
+};
```

Comment:
rename this hook. The current name is confusing. 
like useUserCanAddThreadInBlackoutDate